src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenInParametersTests.cs

```diff
@@ -1459,6 +1459,41 @@ .locals init (int V_0)
 }");
         }
 
+        [Fact]
+        public void InParamEnum()
+        {
+            var text = @"
+class Program
+{
+    static void Main()
+    {
+        Enum1 v = Enum1.Value1;
+        System.Console.WriteLine(M(v));
+    }
+
+    static string M(in Enum1 x) => x.ToString();
```

Comment:
Would be good to also have a test calling it on a `readonly` field of type `Enum1`, just to ensure we are catching these other cases of readonly references too.